src/client.es6.js

```diff
@@ -57,7 +57,7 @@ class ClientApp extends App {
   }
 
   fullPathName () {
-    return document.location.pathname + document.location.search;
+    return this.getState('url');
```

Comment:
when you load the client js it immediately calls `app.render(app.fullPathName(), ...)`.  since the url is rewritten what the page renders (an error page) and what the client would render (the document location) could be entirely different.  using the app's url state ensures both the client and the server are in sync on what the url of the route is.